tendermint/src/rpc/id.rs

```diff
@@ -29,3 +29,27 @@ impl Id {
         Id::Str(uuid.to_string())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::test::test_serialization_roundtrip;
+
+    #[test]
+    fn round_tripping_id() {
+        let str = r#""42""#;
+        test_serialization_roundtrip::<Id>(str);
+
+        let str2 = r#""936DA01F-9ABD-4D9D-80C7-02AF85C822A8""#;
+        test_serialization_roundtrip::<Id>(str2);
+
+        let num = r#"42"#;
+        test_serialization_roundtrip::<Id>(num);
+
+        let zero = r#"0"#;
+        test_serialization_roundtrip::<Id>(zero);
+
+        let null = r#"null"#;
+        test_serialization_roundtrip::<Id>(null);
```

Comment:
@greg-szabo didn't you say that `null` doesn't roundtrip properly? Or did I dream this? It does in fact roundtrip.

Reply:
I said that but that's only the case in a JSON->Object->JSON roundtrip. Test_serialization does an Object->JSON->Object roundtrip, which is fine.

Reply:
Still confusing because `test_serialization_roundtrip` actually does:
JSON->Object->JSON->Object

Reply:
Yes, but we don't compare the JSON objects, as far as I understand. Comparing the Rust objects is fine.

`None` is `None`, whether it came from an empty string or a `null` value. The same is not true in JSON.

Reply:
Thanks for clarifying!